### ISSUE TITLE
Allow standalone set operations (and ValuesExpression) without a wrapping SelectExpression

### ISSUE THREAD

**Issue by roji** [dotnet/efcore#32890]
Our SQL query tree architecture currently represents each set operation as wrapped by a SelectExpression; in SQL generation, if the SelectExpression is detected to be a simple wrapper with no other clauses ([code](https://github.com/dotnet/efcore/blob/main/src/EFCore.Relational/Query/QuerySqlGenerator.cs#L288)), the SelectExpression isn't outputted.

We should change our SQL tree to exactly match the SQL it will output, allowing bare set operations/ValuesExpressions to appear directly as a table in a select expression (and even as the top-level node in the tree).

Having the SelectExpression wrapper in the tree causes some complications, such as having to manage aliases for SelectExpressions that will never actually appear in the output, etc.

**Comment by ajcvickers** [comment]
> we'll likely need to empty SelectExpression entirely out of private state

How the mighty fall...

**Comment by roji** [comment]
@Charlieface I completely agree with your comments - the current design is problematic in many ways and indeed hampers our ability to build features and have more extensibility. SelectExpression really needs to become a plain immutable expression with no actual logic in it.

This is something I've started working on but it's quite a huge rearchitecting effort that will take a long time; I hope to do substantial progress on this in 10.

BTW CTE support - including recursive - is also something I hope we'll implement for 10.
